python/semantic_kernel/connectors/memory/azure_cosmos_db/azure_cosmos_db_no_sql_base.py

```diff
@@ -0,0 +1,99 @@
+from azure.cosmos.exceptions import CosmosResourceNotFoundError
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.memory.azure_cosmos_db.azure_cosmos_db_no_sql_settings import AzureCosmosDBNoSQLSettings
+from semantic_kernel.connectors.memory.azure_cosmos_db.utils import CosmosClientWrapper, DefaultAzureCredentialWrapper
+from semantic_kernel.exceptions.memory_connector_exceptions import (
+    MemoryConnectorInitializationError,
+    MemoryConnectorResourceNotFound,
+)
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+
+@experimental_class
+class AzureCosmosDBNoSQLBase(KernelBaseModel):
+    """An Azure Cosmos DB NoSQL collection stores documents in a Azure Cosmos DB NoSQL account."""
+
+    cosmos_client: CosmosClient
+    database_name: str
+    # If create_database is True, the database will be created
+    # if it does not exist when an operation requires a database.
+    create_database: bool
+
+    def __init__(
+        self,
+        database_name: str,
+        url: str | None = None,
+        key: str | None = None,
+        cosmos_client: CosmosClient | None = None,
+        create_database: bool = False,
+        **kwargs,
+    ):
+        """Initialize the AzureCosmosDBNoSQLBase.
+
+        Args:
+            database_name (str): The name of the database. The database may not exist yet.
+                                 If it does not exist, it will be created when the first collection is created.
+            url (str): The URL of the Azure Cosmos DB NoSQL account. Defaults to None.
+            key (str): The key of the Azure Cosmos DB NoSQL account. Defaults to None.
+            cosmos_client (CosmosClient): The custom Azure Cosmos DB NoSQL client whose lifetime is managed by the user.
+                                          Defaults to None.
+            create_database (bool): If True, the database will be created if it does not exist.
+                                    Defaults to False.
+            kwargs: Additional keyword arguments.
+        """
+        try:
+            cosmos_db_nosql_settings = AzureCosmosDBNoSQLSettings.create(url=url, key=key)
+        except ValidationError as e:
+            raise MemoryConnectorInitializationError("Failed to validate Azure Cosmos DB NoSQL settings.") from e
+
+        if cosmos_client is None:
+            if cosmos_db_nosql_settings.key is not None:
+                cosmos_client = CosmosClientWrapper(
+                    str(cosmos_db_nosql_settings.url), credential=cosmos_db_nosql_settings.key.get_secret_value()
+                )
+            else:
+                cosmos_client = CosmosClientWrapper(
+                    str(cosmos_db_nosql_settings.url), credential=DefaultAzureCredentialWrapper()
+                )
+
+        super().__init__(
+            database_name=database_name,
```

Comment:
should the database name be part of settings?

Reply:
I am open to both either way.